### PR TITLE
Migrate pallet-template to pallet attribute macro

### PR DESCRIPTION
**Pull request by g2udevelopment**
Part of #7882.

Migrates the `Template` pallet in node-template to the new attribute macro's. Tried to keep the documentation as is.

Because it is a template ea example pallet I don't think there are storage changes to think of. If needed I can help with modifying the tutorials if needed and when this is pushed to substrate-developers-hub.

@thiolliere

### CODE REVIEW

File: bin/node-template/pallets/template/src/lib.rs [hunk 1]
```diff
@@ -4,100 +4,101 @@
...
-// Pallets use events to inform users when important changes are made.
-// https://substrate.dev/docs/en/knowledgebase/runtime/events
-decl_event! {
-	pub enum Event<T> where AccountId = <T as frame_system::Config>::AccountId {
+    // The pallet's runtime storage items.
+    // https://substrate.dev/docs/en/knowledgebase/runtime/storage
+    #[pallet::storage]
+    #[pallet::getter(fn something)]
+    // Learn more about declaring storage items:
+	// https://substrate.dev/docs/en/knowledgebase/runtime/storage#declaring-storage-items
```
Comment (bkchr) [inline bin/node-template/pallets/template/src/lib.rs:36]: You are mixing here spaces and tabs.

Comment (g2udevelopment) [reply bin/node-template/pallets/template/src/lib.rs:36]: Fixed the indentations.

File: bin/node-template/pallets/template/src/lib.rs [hunk 2]
```diff
@@ -4,98 +4,99 @@
...
+	// Pallets use events to inform users when important changes are made.
+	// https://substrate.dev/docs/en/knowledgebase/runtime/events
+	#[pallet::event]
+	#[pallet::metadata(T::AccountId = "AccountId")]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
 		/// Event documentation should end with an array that provides descriptive names for event
 		/// parameters. [something, who]
-		SomethingStored(u32, AccountId),
+		SomethingStored(u32,T::AccountId),
```
Comment (thiolliere) [inline bin/node-template/pallets/template/src/lib.rs:47]:
```suggestion
		SomethingStored(u32, T::AccountId),
```

File: bin/node-template/pallets/template/src/lib.rs [hunk 3]
```diff
@@ -4,98 +4,99 @@
...
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	// The pallet's runtime storage items.
+	// https://substrate.dev/docs/en/knowledgebase/runtime/storage
+	#[pallet::storage]
+	#[pallet::getter(fn something)]
+	// Learn more about declaring storage items:
+	// https://substrate.dev/docs/en/knowledgebase/runtime/storage#declaring-storage-items
+	pub type Something<T> = StorageValue<_,u32>;
```
Comment (thiolliere) [inline bin/node-template/pallets/template/src/lib.rs:37]:
```suggestion
	pub type Something<T> = StorageValue<_, u32>;
```

**Review by thiolliere — APPROVED** [review]
thanks, looks good to me, would be greate to also make use of construct_runtime in the tests instead of impl_out_* but we can do in another PR

**Comment by g2udevelopment** [comment]
> thanks, looks good to me, would be greate to also make use of construct_runtime in the tests instead of impl_out_* but we can do in another PR

Thanks, I will convert the tests as well in a separate PR.

**Review by bkchr — APPROVED** [review]

**Pull request merged**